### ISSUE TITLE
Mnasnet - Channels first data format issue

### ISSUE THREAD

**Issue by pidajay** [tensorflow/tpu#417]
Using channels first data format causes the script to fail because of a number of reasons

1. The pre-processing steps assumes the data will always be channels last (even after the data format is set to channels first).
2. There are also several layers in the model (both efficient net and mnasnet) that don't accept the data format (and hence assumes channels last). For eg. Conv2D.
After fixing 1 and 2 I can get the model to run somewhat but fails to run for multi-gpu cases.

This issue was originally raised here but got closed prematurely - https://github.com/tensorflow/tpu/issues/410

**Comment by saberkun** [comment]
Thanks for reporting. 
Yes, I just fixed the bugs inside the code to support channels_first convolution layout.
As TPU utilizes XLA compiler, the memory layout will be optimized.

**Comment by pidajay** [comment]
Awesome thanks! Just curious if you get to test the code on GPUs? If so what is the performance like?

**Comment by saberkun** [comment]
@pidajay, Yes, I tested with an cuda smoke test on GPU like running a few steps. I am afraid we do not have bandwidth/resource to really look into GPU performance. If you have any suggestion or finding, please let us know. 
What if you feed with channels_last input, the model_fn has a transpose to change channels_last to channels_first.
I would suggest something like creating your own branch and experiment with GPU xla and fp16 training. 
Thanks
